polyfill/lib/yearmonth.mjs

```diff
@@ -1,126 +1,113 @@
+    const duration = ES.GetIntrinsic('%Temporal.duration%')(duration);
+    let { year, month } = this;
+    let { years, months, days, hours, minutes, seconds, milliseconds, microseconds, nanoseconds } = duration;
+    if (days || hours || minutes || seconds || milliseconds || microseconds || nanoseconds)
+      throw new RangeError('invalid duration');
+    ({ year, month } = ES.SubtractDate(year, month, 1, years, months, 0, disambiguation));
+    ({ year, month } = ES.BalanceYearMonth(year, month));
+    return new YearMonth(year, month);
+  }
+  difference(other) {
+    other = ES.GetIntrinsic('%Temporal.yearmonth%')(other);
+    const [one, two] = [this, other].sort(DateTime.compare);
+    let years = two.year - one.year;
+    let months = two.month - one.month;
+    if (months < 0) {
+      years -= 1;
+      months += 12;
+    }
+    const Duration = ES.GetIntrinsic('%Temporal.Duration%');
+    return new Duration(years, months);
+  }
+  toString() {
+    let year = ES.ISOYearString(GetSlot(this, YEAR));
+    let month = ES.ISODateTimePartString(GetSlot(this, MONTH));
+    let resultString = `${year}-${month}`;
+    return resultString;
+  }
+  toLocaleString(...args) {
+    return new Intl.DateTimeFormat(...args).format(this);
+  }
+  withDay(day, disambiguation = 'constrain') {
+    const year = GetSlot(this, YEAR);
+    const month = GetSlot(this, MONTH);
+    const Date = ES.GetIntrinsic('%Temporal.Date%');
+    return new Date(year, month, day, disambiguation);
   }
-  const Duration = ES.GetIntrinsic('%Temporal.Duration%');
-  return new Duration(years, months);
-};
-YearMonth.prototype.toString = YearMonth.prototype.toJSON = function toString() {
-  let year = ES.ISOYearString(GetSlot(this, YEAR));
-  let month = ES.ISODateTimePartString(GetSlot(this, MONTH));
-  let resultString = `${year}-${month}`;
-  return resultString;
-};
-YearMonth.prototype.toLocaleString = function toLocaleString(...args) {
-  return new Intl.DateTimeFormat(...args).format(this);
-};
-YearMonth.prototype.withDay = function withDay(day, disambiguation = 'constrain') {
-  const year = GetSlot(this, YEAR);
-  const month = GetSlot(this, MONTH);
-  const Date = ES.GetIntrinsic('%Temporal.Date%');
-  return new Date(year, month, day, disambiguation);
-};
-
-YearMonth.fromString = function fromString(isoStringParam) {
-  const isoString = ES.ToString(isoStringParam);
-  const match = DATE.exec(isoString);
-  if (!match) throw new RangeError('invalid date string');
-  const year = ES.ToInteger(match[1]);
-  const month = ES.ToInteger(match[2]);
-  return new YearMonth(year, month, 'reject');
-};
-YearMonth.compare = function compare(one, two) {
-  if (one.year !== two.year) return one.year - two.year;
-  if (one.month !== two.month) return one.month - two.month;
-  return 0;
-};
 
-Object.defineProperty(YearMonth.prototype, Symbol.toStringTag, {
-  value: 'Temporal.YearMonth'
-});
+  static fromString(isoStringParam) {
```

Comment:
Should this method be removed, superseded by from()? Or should we maintain both?

Reply:
No, the explicit string from as a counterpoint `toString` should be kept.

Reply:
Why is that crucial? If you want that, you’d ensure you have a string, and then call the from function.

Reply:
Because there is then no way to force an error when you pass in something that is NOT a string.

It's not about the happy path where `Absolute.from(value)` receives a string, it's about requiring an iso-string to be passed in. Even if I typecheck the string to be a string, it could be `"123"` which would fail the parsing, but would then be interpreted as `+"123"` which is milliseconds since epoch.